Renaming the name of the controlClass attribute for form fields

korenevskiy opened the pull request:
I apologize.
I believe that the name of the newly added attribute for the fields is not quite correct.
Calling it the parent class. This is logical, but it is not clear which parent it refers to.
At first glance, you might think that this is the parent of the field. But this is not the case.
After thinking about it, you might think that this is the parent of the control, but this is also not the case. Since the control itself is the target.
![img-2021-03-09-13-35-32](https://user-images.githubusercontent.com/6898474/110778512-7f171080-8273-11eb-994f-06edd7c9da7e.png)
https://github.com/joomla/joomla-cms/pull/32488
The author @ciar4n  of the attribute agreed to replace the attribute name.

korenevskiy committed:
Replace parentclass to controlclass.

Replace parentclass to controlclass.
This is necessary for a more understandable associative purpose of this attribute.

Quy commented on libraries/src/Form/FormField.php:
```diff
@@ -1351,7 +1351,7 @@ protected function getLayoutData()
 			'dataAttribute'  => $this->renderDataAttributes(),
 			'dataAttributes' => $this->dataAttributes,
-			'parentclass'    => $this->parentclass,
+			'controlclass'    => $this->controlclass,
```
```suggestion
			'controlclass'   => $this->controlclass,
```

ciar4n commented:
> The author @ciar4n of the attribute agreed to replace the attribute name.

Not exactly true 😄 ...  I just said I wasn't going to apply your suggestion to a PR that had already gone to RTC.

korenevskiy commented:
> > The author @ciar4n of the attribute agreed to replace the attribute name.
> 
> Not exactly true 😄 ...  I just said I wasn't going to apply your suggestion to a PR that had already gone to RTC. 

but you put a thumbs-up emoji there. 👍

ceford commented:
I have tested this item :white_check_mark: successfully on 6901dbc973018dabe98c9eaa4deb66f13419552d<br /><br />I looked at the diff, applied the patch and searched for parentclass and controlclass. I think it is OK.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32645">issues.joomla.org/tracker/joomla-cms/32645</a>.</sub>

dgrammatiko commented:
I think if you dislike the `parentclass` an alternative **logical** value could be `containerclass` and **not** `controlclass`.

korenevskiy committed:
[J4-Template] Add position without parent tag

**`<jdoc:include type="modules" name="top" style="default" class="grid-child" />`**

and

**`<jdoc:include type="modules" name="bottom" style="default" class="grid-child" />`**

Friends, please consider the offer to add 2 more positions to the template.
What is it for?
Answer: These fields will have little value for users who use the Cassiopeia template as it is.
BUT!, While a lot of developers use the Cassiopeia template as a basis for redefining new styles. I do it myself.
The fact is that the Bootstrap is already built into Cassiopeia, the grid, controlls, margins, and overrides of default styles are already configured. You don't need to do anything about it. Therefore, it is logical for any templates to use a ready-made Cassiopeia template to redefine its styles.
This is to add a background image, change the color. Simple as that.
But at the same time, you need to have your entire position without the parent tag.
so that you can place as many modules as you want in this position, each of which will have its own grid for mobile screens.
```
<?php if ($this->countModules('bottom-b')): ? >
	<div class="grid-child container-bottom-b">
		<jdoc:include type="modules" name="bottom-b" style="card" />
	</div>
<?php endif; ?>
```
The container-bottom-b class and any other container class have their own built-in grid. What causes problems when the grid is not needed at all in the container class.

`<jdoc:include type="modules" name="bottom" style="default" class="grid-child" />`

**`full-width`** or **`containier-full`** (for old template as 'protostar') of **`wrapper-fluid`**

In addition, I suggest that you be able to use your own class in `<jdoc:include type="modules" name="bottom" style= "default" css="grid-child"/>`.
Thus, the position will be automatically placed in the template grid in the center of the content.
However, if the class prefix is added by the user, I suggest excluding the position class altogether. This will allow the user to add the full-grid class themselves, or to add the `grid` class or `card` class themselves.

for example, you need to make a map at the bottom of the site in the full width of the site with the route. But how to do this if the container is bounded by the central column of the grid.

korenevskiy committed:
Merge branch '4.0-dev' of https://github.com/joomla/joomla-cms into 4.0-dev

korenevskiy committed:
Revert "[J4-Template] Add position without parent tag"

This reverts commit ba46c53732e6ba0b99b73468e49fe201b7d05f3a.

korenevskiy committed:
Merge branch '4.0-dev' of https://github.com/korenevskiy/joomla-cms into 4.0-dev

korenevskiy committed:
[J4-Template] Add position without parent tag

**`<jdoc:include type="modules" name="top" style="default" class="grid-child" />`**

and

**`<jdoc:include type="modules" name="bottom" style="default" class="grid-child" />`**

Friends, please consider the offer to add 2 more positions to the template.
What is it for?
Answer: These fields will have little value for users who use the Cassiopeia template as it is.
BUT!, While a lot of developers use the Cassiopeia template as a basis for redefining new styles. I do it myself.
The fact is that the Bootstrap is already built into Cassiopeia, the grid, controlls, margins, and overrides of default styles are already configured. You don't need to do anything about it. Therefore, it is logical for any templates to use a ready-made Cassiopeia template to redefine its styles.
This is to add a background image, change the color. Simple as that.
But at the same time, you need to have your entire position without the parent tag.
so that you can place as many modules as you want in this position, each of which will have its own grid for mobile screens.
```
<?php if ($this->countModules('bottom-b')): ? >
	<div class="grid-child container-bottom-b">
		<jdoc:include type="modules" name="bottom-b" style="card" />
	</div>
<?php endif; ?>
```
The container-bottom-b class and any other container class have their own built-in grid. What causes problems when the grid is not needed at all in the container class.

`<jdoc:include type="modules" name="bottom" style="default" class="grid-child" />`

**`full-width`** or **`containier-full`** (for old template as 'protostar') of **`wrapper-fluid`**

In addition, I suggest that you be able to use your own class in `<jdoc:include type="modules" name="bottom" style= "default" css="grid-child"/>`.
Thus, the position will be automatically placed in the template grid in the center of the content.
However, if the class prefix is added by the user, I suggest excluding the position class altogether. This will allow the user to add the full-grid class themselves, or to add the `grid` class or `card` class themselves.

for example, you need to make a map at the bottom of the site in the full width of the site with the route. But how to do this if the container is bounded by the central column of the grid.

korenevskiy committed:
Merge branch 'joomla:4.0-dev' into 4.0-dev

korenevskiy committed:
Merge branch 'joomla:4.0-dev' into 4.0-dev

korenevskiy committed:
Merge branch 'joomla:4.0-dev' into 4.0-dev

korenevskiy committed:
[J4] not working switch RECORD HITS

![image](https://user-images.githubusercontent.com/6898474/128238875-fd884b74-a53b-46c5-b020-ace4ce5abfe4.png)
![image](https://user-images.githubusercontent.com/6898474/128239158-cb814349-0f5c-4a70-9f92-4341000ac344.png)

In the Article settings. The RECORD HITS field is disabled.
But this value is displayed on the site.
The fact is that the disabled RECORD HITS field hides another HITS field.
I believe that it should be like this:
- Either the RECORD HITS field will not hide the HITS field.
- Either disabling the RECORD HITS field will automatically deactivate the HITS field.
![image](https://user-images.githubusercontent.com/6898474/128240189-14b98eb6-6467-415b-af7b-5524bbf04d38.png)

korenevskiy committed:
Merge pull request #1 from korenevskiy/Fix-admin-switcher-ShowOn-Hits

![image](https://user-images.githubusercontent.com/6898474/128238875-fd884b74-a53b-46c5-b020-ace4ce5abfe4.png)
![image](https://user-images.githubusercontent.com/6898474/128239158-cb814349-0f5c-4a70-9f92-4341000ac344.png)

In the Article settings. The RECORD HITS field is disabled.
But this value is displayed on the site.
The fact is that the disabled RECORD HITS field hides another HITS field.
I believe that it should be like this:
- Either the RECORD HITS field will not hide the HITS field.
- Either disabling the RECORD HITS field will automatically deactivate the HITS field.
![image](https://user-images.githubusercontent.com/6898474/128240189-14b98eb6-6467-415b-af7b-5524bbf04d38.png)

Pull Request for Issue # .

HLeithner commented:
This pull request has automatically rebased to 4.2-dev.

korenevskiy committed:
Merge branch '4.2-dev' into 4.0-dev

korenevskiy committed:
Update renderfield.php

korenevskiy committed:
Update card.php

korenevskiy committed:
Update noCard.php

korenevskiy committed:
Update renderfield.php

korenevskiy committed:
Update FormField.php
